### PR TITLE
Add L2 CM tracking

### PR DESCRIPTION
**Pull request by adel-mamin**
This PR is contingent on 
https://github.com/gsmcmullin/piksi_firmware/pull/4 
and 
https://github.com/swift-nav/libswiftnav-private/pull/6

### CODE REVIEW

**Comment by adel-mamin** [comment]
@valeri-atamaniouk Please review.

File: src/sbp_utils.c [hunk 1]
```diff
@@ -294,7 +294,7 @@ void unpack_ephemeris(const msg_ephemeris_t *msg, ephemeris_t *e)
   e->sid              = sid_from_sbp(msg->sid);
   e->kepler.iode      = msg->iode;
   e->kepler.iodc      = msg->iodc;
-  e->fit_interval     = 4 * 60 * 60; /* TODO: this is a work around until SBP updated */
+  e->fit_interval     = (u8)(4 * 60 * 600); /* TODO: this is a work around until SBP updated */
```
Comment (valeri-atamaniouk) [inline src/sbp_utils.c]: Why u8 cast if the number is 16 bit at least?

File: src/track/track_gps_l2cm.c [hunk 2]
```diff
@@ -0,0 +1,473 @@
...
+    log_info_sid(channel_info->sid, "PLL stress");
+    tracker_ambiguity_unknown(channel_info->context);
+  }
+
+  /* Run the loop filters. */
+
+  /* Output I/Q correlations using SBP if enabled for this channel */
+  tracker_correlations_send(channel_info->context, cs);
+
+  correlation_t cs2[3];
```
Comment (valeri-atamaniouk) [inline src/track/track_gps_l2cm.c:530]: I believe the same code is in L1CA. Why not to make a helper function?

File: src/track/track_gps_l2cm.c [hunk 3]
```diff
@@ -0,0 +1,473 @@
...
+  if (l->coherent_ms != L2C_COHERENT_INTEGRATION_TIME_MS) {
+    log_error("Invalid coherent integration length for L2CM: %" PRIu8,
+              l->coherent_ms);
+    return false;
+  }
+  /* Successfully parsed the input. Save to memory. */
+  strncpy(s->addr, val, s->len);
+  if (s->len > 0) {
+    char *ptr = (char*) s->addr;
+    ptr[s->len - 1] = '\0';
```
Comment (valeri-atamaniouk) [inline src/track/track_gps_l2cm.c:605]: If len == 0, s->addr is uninitialized.

File: src/track/track_gps_l2cm.c [hunk 4]
```diff
@@ -0,0 +1,473 @@
...
+
+/** Parse a string describing the tracking loop filter parameters into
+ *  the loop_params_stage struct.
+ *
+ * \param s Settings structure provided to store the input string.
+ * \param val The input string to parse.
+ * \retval true Success
+ * \retval false Failure
+ */
+static bool parse_loop_params(struct setting *s, const char *val)
```
Comment (valeri-atamaniouk) [inline src/track/track_gps_l2cm.c:575]: This function is very similar in l1ca - please make common

Comment (valeri-atamaniouk) [inline src/track/track_gps_l2cm.c:575]: Looks good, some modified files need copyright updates

**Comment by adel-mamin** [comment]
This PR is obsoleted by https://github.com/swift-nav/piksi_firmware/pull/672

**Pull request closed without merging**